src/slime_parser.peg.eex

```diff
@@ -1,32 +1,33 @@
-document <- doctype? tags_list crlf* eof;
+document <- doctype? tags? eof;
 
-doctype <- space? 'doctype' space (!eol .)+ eol;
+doctype <- space? 'doctype' space name:(!eol .)+ eol;
 
-% TODO: We can parse nested structure like this:
-% tags_list <- tag (crlf tag)*;
-% tag <- verbatim_text / space? tag_item (crlf empty_lines:(crlf space?)* indent tags_list dedent)? / '' &.;
-% Also remove dedent from eol definition and modify transforms accordingly
-tags_list <- tag (crlf indent? empty_lines:(space? crlf indent?)* tag:tag dedent*)*;
+tag <- comment / verbatim_text / tag_item;
+tag_item <- space? (embedded_engine / inline_html / code / html_tag);
 
-% NOTE: Leading whitespace is handled separately for comments and verbatim text
-tag <- comment / verbatim_text / space? tag_item / blank:(space? &eol);
+tags <- (tag crlf*)+;
+nested_tags <- crlf+ indent tags dedent;
 
-tag_item <- (embedded_engine / inline_html / code / html_tag);
+inline_html <- &'<' text_content nested_tags?;
 
-inline_html <- &'<' text_content;
+html_tag <- inline_tag / simple_tag;
+
+inline_tag <- tag:tag_shortcut spaces:tag_spaces? ':'
+  space children:(inline_tag / simple_tag);
 
-code <- inline:('=' '='? tag_spaces? / '-') space? code:code_lines;
-code_lines <- code_line / code_line_with_brake crlf (indent code_lines dedent / code_lines);
+simple_tag <- tag:tag_shortcut spaces:tag_spaces space?
+  content:attributes_with_content children:nested_tags?;
```

Comment:
There are some invalid test cases, currently supported by accident, in ruby-slim tag with inline content could not have nested children tags, (if there are nested lines, they treated as inline content continuation). In some cases it as already not handled by slime, so I think it is time to fix this behaviour, or at least have a parsing error for this case.
For example:
```
div test
  test
```
in slim renders to
```
<div>test
test</div>
```